recipes/qat_distributed.py

```diff
@@ -127,6 +127,12 @@ def __init__(self, cfg: DictConfig) -> None:
         self._log_every_n_steps = cfg.get("log_every_n_steps", 1)
         self._log_peak_memory_stats = cfg.get("log_peak_memory_stats", False)
 
+        if self._log_peak_memory_stats and self._device.type == "cuda":
```

Comment:
Thoughts on whether we actually need this? I realise we kind of fail "silently" at the moment by just not logging if we aren't running on CUDA https://github.com/pytorch/torchtune/blob/17ba37d281548e5d60215f741888066717ad5b3e/recipes/lora_finetune_single_device.py#L716-L719)

@joecummings @felipemello1 

As-is we're now doing duplicating this check, which isn't super clean. Personally I'd rather just make the check in the relevant logging util - but don't have to block on this.

Reply:
let me get back to you on this @krammnic . Thanks for making all of these changes! :)

Reply:
I'm not really see the problem in 2 partially duplicating checks. The point is, that if we have such logic in train:

```
log_dict = {
    "loss": loss_to_log,
    "lr": self._optimizer.param_groups[0]["lr"],
    "tokens_per_second_per_gpu": num_tokens / time_per_step,
}
if self._log_peak_memory_stats:
    log_dict.update(
        training.get_memory_stats(device=self._device)
    )
self._metric_logger.log_dict(
    log_dict,
    step=self.global_step,
)
```

We can't do anything better, can we? Check in `__init__` is about cuda and logging(once). Check in train probably should not be about "cuda"(there is no use case) and not about logging. I'm not sure if this should be in `_metric_logger` either